forte/pipeline.py

```diff
@@ -798,6 +812,16 @@ def add(
             # This will ask the job to keep a copy of the gold standard.
             self.evaluator_indices.append(len(self.components))
 
+        if ref_name is None:
+            self._ref_names[ref_name] = []
+            self._ref_names[ref_name].append(len(self.components))
```

Comment:
So when the ref_name is None, we should simply not store it. The behavior is weird if we query the component with a "None" as a key.